enhancements/ingress/path-rewriting.md

```diff
@@ -0,0 +1,122 @@
+- [ ] Test plan is defined
+- [ ] Graduation criteria for dev preview, tech preview, GA
+- [ ] User-facing documentation is created in [openshift-docs](https://github.com/openshift/openshift-docs/)
+
+## Summary
+
+Some Ingress controller implementations support doing path rewriting. This is
+used to modify the path in the HTTP request as it gets proxied:
+
+`http://example.com/path` ---> `http://example.com/`
+
+This is a proposal to support path rewriting for Routes as well.
+
+## Motivation
+
+To use non wildcard SSL certificates we want to direct traffic based on the path
+rather then hostname (e.g. `http://example.com/app` instead of
+`http://app.example.com/`).
+
+We want to remove the path from the HTTP requests (`http://example.com/path` -->
+`http://example.com/`) because the upstream services are serving on /.
+
+There is an open issue about that: <https://github.com/openshift/origin/issues/20474>
+
+## Proposal
+
+### User Stories
+
+[CodeReady Workspaces](https://developers.redhat.com/products/codeready-workspaces/overview)
+is a controller that dynamically creates developers workspaces (web IDEs). A
+workspace is a web applications with N microservices exposing N routes (N
+depends on the workspace but is usually ~5).
+
+When a developer starts his workspace, N routes are created. When he stops it
+the N routes are deleted.
+
+**Wildcard certificates problems**
+
+By default workspaces routes have URLs with random subdomains like
+https://<random-part>.example.com. We create a lots of these routes and
+that's not an issue if the customer is allowed to use a wildcard TLS
+certificate (https://*.example.com).
+
+But not all customers can use wildcard certs. For that reason we now support
+fixed domain scenario using URLs with a fixed hostname and a variable path:
+https://example.com/<random-part>/. We have been able to implement this
+single-host option because upstream Ingress controllers support path rewriting.
+For instance with the nginx controller adding the following annotation does
+the trick:
+
+`traefik.ingress.kubernetes.io/rewrite-target: /`
+`nginx.ingress.kubernetes.io/rewrite-target: /$1`
+
+We would like to support these annotations for Routes as well.
+We don't need support of regex path matching as `Route.spec.path` is not using regex as well.
+
+**Route-specific annotations**
+
+| Variable	| Description | Environment variable used as default |
+| --------- | ----------- | ------------------------------------ |
+| router.openshift.io/rewrite-target | Sets the rewrite path of the request on the backend.| |
+
+**Route-rewrite examples**
+
+| Route.spec.path	| Request path | Rewrite target | Output |
+| --------------- | ----------- | --------------- | ------ |
+| /foo | /foo | / | / |
+| /foo | /foo/ | / | / |
+| /foo | /foo/bar | / | /bar |
+| /foo | /foo/bar/ | / | /bar/ |
+| /foo | /foo | /bar | /bar |
+| /foo | /foo/ | /bar | /bar/ |
+| /foo | /foo/bar | /baz | /baz/bar |
+| /foo | /foo/bar/ | /baz | /baz/bar |
```

Comment:
Some applications deals differently when there is a trailing `/` in the path at the end. 

Are we saying we're going to remove the trailing `/` from now on?

(BTW, I've got no ill intention, I'm just asking as I'm going to be a user, please don't hate me XD)

Reply:
(It seems inconsistent with line 91)

Reply:
Thanks for the catch. I did not recopy the right stuff.
Now I will remove the content from there and link it to https://github.com/openshift/openshift-docs/pull/22021

Reply:
Thanks for coming back to me :) I really appreciate.